docs/source/en/model_doc/zamba.md

```diff
@@ -0,0 +1,53 @@
+# Zamba
+
+Zamba is a large language model (LLM) trained by Zyphra, and made available under an Apache 2.0 license. Please see the [Zyphra Hugging Face](https://huggingface.co/Zyphra) repository for model weights.
+
+
+## Model details
+
+Zamba-7B-v1 is a hybrid between state-space models (Specifically [Mamba](https://github.com/state-spaces/mamba)) and transformer. Zamba consists of a Mamba backbone with a shared transformer block every 6 Mamba layers. We came to this architecture after a series of ablations at small scales, where we found it proved highly effective at combining the inference efficiency of Mamba with the minimal amount of attention required to maintain performance and expressivity. Zambaand was trained using next-token prediction and uses the [Mistral v0.1 tokenizer](https://huggingface.co/mistralai/Mistral-7B-v0.1).  Zamba-7B-v1 was pre-trained on 1T tokens of text and code data. We then performed an annealing phase over 50B high quality tokens.
+
+<img src="https://github.com/Zyphra/HF-zamba/blob/main/zamba-arch.png" width=40% height=40% />
+
+
+## Quick start
+
+### Presequities
+
+Zamba requires you use `transformers` version 4.39.0 or higher:
+```bash
+pip install transformers>=4.39.0
+```
+
+In order to run optimized Mamba implementations on a CUDA device, you first need to install `mamba-ssm` and `causal-conv1d`:
+```bash
+pip install mamba-ssm causal-conv1d>=1.2.0
+```
+
+You can run the model not using the optimized Mamba kernels, but it is **not** recommended as it will result in significantly higher latency. 
+
+To run on CPU, please specify `use_mamba_kernels=False` when loading the model using ``AutoModelForCausalLM.from_pretrained``.
```

Comment:
Is this true? I don't think we do support that argument in `from_pretrained` currently

Reply:
Yes, we tested this. `use_mamba_kernels` is a parameter in `ZambaConfig` that defaults to `True`. We tested that passing `use_mamba_kernels=False` in `from_pretrained` sets the forward pass to the one using the pure Python implementation.

Reply:
Indeed, it should be dealt here: https://github.com/huggingface/transformers/blob/6b11f89c6b3d15655a2264df73e453cf9692c62e/src/transformers/modeling_utils.py#L3229C19-L3229C25 all good !